libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -53,30 +38,30 @@ internal class MapboxDirectionsSession(
 
     override fun setRoutes(
         routes: List<NavigationRoute>,
+        ignoredRoutes: List<IgnoredRoute>,
         setRoutesInfo: SetRoutes,
     ) {
         this.initialLegIndex = setRoutesInfo.initialLegIndex()
-        if (routesInitialized && this.routes.isEmpty() && routes.isEmpty()) {
+        if (routesUpdatedResult?.navigationRoutes?.isEmpty() == true && routes.isEmpty()) {
```

Comment:
seems like it breaks previous changes from https://github.com/mapbox/mapbox-navigation-android/pull/6097/files#r928965604 . If twice set empty routes observers are not triggered

Reply:
`routesInitialized && this.routes.isEmpty()` is the same as `routesUpdatedResult != null && routesUpdatedResult.navigationRoutes.isEmpty()` which is the same as `routesUpdatedResult?.navigationRoutes?.isEmpty() == true`.

Reply:
Oh, we already have test for this: https://github.com/mapbox/mapbox-navigation-android/pull/6745/files#diff-8b7f524d461ddbed81d505886c466453ba27ac24053a345415fee3e6b1e6d3abR354

@RingerJK , does it cover the case you're asking about?